### PR TITLE
preparing 8.3.0 pre

### PR DESCRIPTION
**Pull request by nfrisby**
# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by nfrisby** [comment]
I re-requested review from @Jimbo4350 , because I pushed to this branch a few minutes ago not realizing that he had Approved the PR.

**Comment by nfrisby** [comment]
The CI is failing because:

```
$ cabal build all --dry-run --minimize-conflict-set
  shell: /usr/bin/bash -e {0}
  env:
    CABAL_CACHE_VERSION: 2023-07-28
    SECP_CACHE_VERSION: 2022-12-30
    SECP256K1_REF: ac83be33d0956faf6b7f61a60ab524ef7d6a473a
    PKG_CONFIG_PATH: /usr/lib/pkgconfig:/usr/local/lib/pkgconfig
    LD_LIBRARY_PATH: /usr/local/lib
    SECP256K1_PREFIX: /usr/local
Resolving dependencies...
Error: cabal: Could not resolve dependencies:
[__0] trying: cardano-node-8.2.1 (user goal)
[__1] next goal: cardano-crypto-class (dependency of cardano-node)
[__1] rejecting: cardano-crypto-class-2.1.2.0 (conflict: pkg-config package
libblst-any, not found in the pkg-config database)
```

**Comment by nfrisby** [comment]
I rebased in order to pick up John's https://github.com/input-output-hk/cardano-node/pull/5445

**Comment by nfrisby** [comment]
Most recent CI failure was due to the Conway genesis file being too much of a stub; the newer ledger requires more structure be present than did the previous ledger, so the stub needs to be fleshed out.

I just pushed up a TOSQUASH commit that does that, and also updates the file's hash in the other config files.

FYI @disassembler @Jimbo4350 -- in case that change is disruptive to some config outside of the repository. Also: in case a non-stub version of this file already exists somewhere?

**Comment by nfrisby** [comment]
Fixing the Conway genesis file did not resolve the remaining CI errors. I've asked for help via Slack.

**Comment by angerman** [comment]
```
genesis_keepalive for Hydra
workbench:  genesis:  new one:  profile /nix/store/ld2fk892bmsmb6pvxdq9qhyhlvzv5qq9-profile-ci-test-bage.json node_specs /nix/store/alhgkx5jhilylwkmx6jbaybswq4njn2j-node-specs-ci-test-bage.json dir /nix/store/nbgw0im89kavchgl2bb08234bf4998kl-workbench-profile-genesis-cache-ci-test-bage cache_key k2-0kD-0kU-31cad58 cache_key_input { "delegators": 0, "dense_pool_density": 1, "funds_balance": 10000000000000, "n_bft_hosts": 0, "n_dense_hosts": 0, "n_pools": 2, "network_magic": 42, "per_pool_balance": 1000000000000000, "pool_coin": 1000000000000000, "utxo_stuffed": 0 }
Command failed: genesis create  Error: Error while decoding Shelley genesis at: /nix/store/nbgw0im89kavchgl2bb08234bf4998kl-workbench-profile-genesis-cache-ci-test-bage/genesis.conway.spec.json Error: Error in $: key "poolVotingThresholds" not found
workbench:  failed: cardano-cli genesis create --genesis-dir /nix/store/nbgw0im89kavchgl2bb08234bf4998kl-workbench-profile-genesis-cache-ci-test-bage/ --supply 2010000000000000 --testnet-magic 42 --gen-genesis-keys 0 --gen-utxo-keys 1
```

So the genes file still seems to be missing something. 😕

**Comment by angerman** [comment]
looking closer at this, looks like `cardano-cli` needs to be updated?
```
workbench:  failed: cardano-cli genesis create --genesis-dir /nix/store/nbgw0im89kavchgl2bb08234bf4998kl-workbench-profile-genesis-cache-ci-test-bage/ --supply 2010000000000000 --testnet-magic 42 --gen-genesis-keys 0 --gen-utxo-keys 1
```
that's the command that generates the 
```
/nix/store/nbgw0im89kavchgl2bb08234bf4998kl-workbench-profile-genesis-cache-ci-test-bage/genesis.conway.spec.json
```
which then fails with
```
Error: Error in $: key "poolVotingThresholds" not found
```

File: nix/haskell.nix [hunk 1]
```diff
@@ -122,7 +127,7 @@ let
           ({ lib, pkgs, ... }: {
             # Use the VRF fork of libsodium
             packages.cardano-crypto-praos.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf ] ];
-            packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
+            packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 pkgs.libblst ] ];
```
Comment (angerman) [inline nix/haskell.nix:130]: I'm not sure we need these anymore.

**Comment by angerman** [comment]
@nfrisby I'm sorry I don't fully see through what's broken here. The prop tests that fail also seem to be related to cardano-cli. Though I haven't found out what that really is. I hope I can leave this with you for now. Please ping me if you run into technical CI issues again.

**Comment by angerman** [comment]
Some further investigation by @andreabedini showed that it's the workbench:

https://github.com/input-output-hk/cardano-node/blob/70dde196367e86c977fbe72167f4eb303b3ffa88/nix/workbench/genesis/genesis.sh#L164C4-L164C4

**Review by mgmeier — APPROVED** [review]
LGTM. Thank you @nfrisby

**Review by disassembler — APPROVED** [review]

**Review by carbolymer — APPROVED** [review]

**Pull request merged**